sdk/azcore/policy_token_refresh.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+)
+
+const (
+	bearerTokenPrefix = "Bearer "
+)
+
+// DO NOT USE. This is for internal SDK usage. The TokenRefreshProcessor is used to provide
+// custom implementations for the bearer token policy.
+type TokenRefreshProcessor interface {
+	// IsZeroOrExpired returns a bool if there is a tenant that needs to be initialized, it also
+	// returns a list of tenants to get a token for
+	IsZeroOrExpired(map[string]time.Time) (bool, []string)
+	// ShouldRefresh returns a bool is a token needs to be refreshed and the list of tenants that
+	// need to be refreshed
+	ShouldRefresh(map[string]time.Time) (bool, []string)
+	// GetToken performs the GetToken call and returns the header needed for authentication
+	GetToken(ctx context.Context, cred TokenCredential, tenants []string, opts TokenRequestOptions) (string, error)
```

Comment:
How about we call this `Refresh()` to match `TokenRefresher`.  It's really not a singular token for the multi-tenant case anyways.

Reply:
Refresh sounds good to me...but maybe should it be `RefreshOrGet()`? Since it isnt always a refresh? First it initializes it

Reply:
I think it likely depends on your perspective (FWIW we have a similar pattern in `go-autorest`).  Maybe it would be better to call the interface `TokenFetcher` and make this method `Fetch()`?

Reply:
Yeah `TokenFetcher` and `Fetch` work...I'll update it. I do think all of these additions to public surface area should be approved by Jeff in a review eventually but this work for me.

Reply:
Done